File: Java/083. Numeros/Numeros.java
import java.util.Scanner;
public class Numeros
{
    public static void main(String[] args)
    {
        Scanner scan = new Scanner(System.in);
        int numero = scan.nextInt();
        int somadiv = 0;
        while (numero >= 0)
        {
            for (int i = 1; i < numero; i++)
            {
                if (numero % i == 0)
                {
                    somadiv += i;
                }
            }
            if (somadiv > numero)
            {
                System.out.println("abundante");
            }
            else if (somadiv < numero)
            {
                System.out.println("deficiente");
            }
            else if (somadiv == numero)
            {
                System.out.println("perfeito");
            }
            numero = scan.nextInt();
            somadiv = 0;               
        }
        scan.close();
    }
}

File: C++/083. Numeros/Numeros.cpp
#include <iostream>
int main()
{
    int n;
    std::cin >> n;
    int somaDiv = 0;
    while (n >= 0) 
    {
        for (int i = 1; i < n; ++i) 
        {
            if (n % i == 0) 
            {
                somaDiv += i;
            }
        }
        if (somaDiv == n)
        {
            std::cout << "perfeito" << std::endl;
        }
        else if (somaDiv > n)
        {
            std::cout << "abundante" << std::endl;
        }
        else
        {
            std::cout << "deficiente" << std::endl;
        }
        somaDiv = 0;
        std::cin >> n;  
    }
    return 0;
}